pkg/provider/consulcatalog/consul_catalog.go

```diff
@@ -48,9 +54,13 @@ type Provider struct {
 	Cache             bool            `description:"Use local agent caching for catalog reads." json:"cache,omitempty" toml:"cache,omitempty" yaml:"cache,omitempty" export:"true"`
 	ExposedByDefault  bool            `description:"Expose containers by default." json:"exposedByDefault,omitempty" toml:"exposedByDefault,omitempty" yaml:"exposedByDefault,omitempty" export:"true"`
 	DefaultRule       string          `description:"Default rule." json:"defaultRule,omitempty" toml:"defaultRule,omitempty" yaml:"defaultRule,omitempty"`
+	ConnectAware      bool            `description:"Enable Consul Connect support." json:"connectAware,omitempty" toml:"connectAware,omitempty" yaml:"connectAware,omitempty"`
```

Comment:
Should we add that when in Consul Connect mode, we _only_ talk to the other connect aware nodes (and we ignore all others), or is that obvious to Consul users?

Reply:
We don't, normal services are still usable as well.

Reply:
To elaborate: setting `ConnectAware` triggers loading the required certs for traefik and then when `traefik.connect=true` as label then it will use connect, otherwise it will do a normal http connection. Unless `ConnectByDefault` is set in which case it accesses all services via connect.

Reply:
Yes, but I'm referring to the fact that in `func (p *Provider) fetchService` if we are in connect mode, we use `p.client.Catalog().Connect` instead of `p.client.Catalog().Service`, which afaiu will result on only listing the connect aware services.
Doesn't it follow from that (and maybe that's where I'm missing something) that factually we'll only ever talk to connect aware peers?

Reply:
Yes, if `fetchService` is called with `connectEnabled=true` (which is a result of having the tag `connect.enabled=true` in that service in the catalog (`p.client.Catalog().Services()` -- see `fetchServices`), then for this one (and only this one) service connect is enforced. But it is still possible that communication to other services from the  catalog happens without connect. `ConnectAware` as provider option is just an initial toggle that controls whether connect can theoretically be used at all.

Does that clear it up a but?

Reply:
I know there's nothing preventing non-connect connections to happen, but I'm not fully convinced it would actually happen, but nevermind. If it ever needs further clarifying, we can do it later on.